File: relationship-manager/notification-service/src/main/java/com/bank/rm/notification/dto/NotificationDtos.java
package com.bank.rm.notification.dto;

import com.bank.rm.common.dto.CommunicationChannel;
import com.bank.rm.common.dto.NotificationStatus;
import com.bank.rm.common.dto.NotificationType;
import java.time.Instant;
import java.util.UUID;

public class NotificationDtos {

    public record SendNotificationRequest(UUID customerId, NotificationType type,
                                           CommunicationChannel channel, String subject,
                                           String body, String recipientAddress) {}

    public record NotificationResponse(UUID notificationId, UUID customerId,
                                        NotificationType type, CommunicationChannel channel,
                                        NotificationStatus status, String subject, Instant createdAt) {}

    private NotificationDtos() {}
}

File: relationship-manager/notification-service/src/main/kotlin/com/bank/rm/notification/dto/NotificationDtos.kt
package com.bank.rm.notification.dto

import com.bank.rm.common.dto.*
import java.util.UUID

data class SendNotificationRequest(
    val customerId: UUID,
    val type: NotificationType,
    val channel: CommunicationChannel,
    val subject: String,
    val body: String,
    val recipientAddress: String,
    val fallbackChannel: CommunicationChannel? = null
)

data class NotificationResponse(
    val id: UUID,
    val customerId: UUID,
    val type: NotificationType,
    val channel: CommunicationChannel,
    val subject: String,
    val status: NotificationStatus,
    val sentAt: String?,
    val externalId: String?
)
